File: app/src/main/java/se/gustavkarlsson/skylight/android/dagger/components/AuroraChanceFragmentComponent.java
package se.gustavkarlsson.skylight.android.dagger.components;

import dagger.Subcomponent;
import se.gustavkarlsson.skylight.android.dagger.modules.replaceable.FragmentRootViewModule;
import se.gustavkarlsson.skylight.android.dagger.scopes.FragmentScope;
import se.gustavkarlsson.skylight.android.gui.activities.main.fragments.aurora_chance.AuroraChanceFragment;

@SuppressWarnings("WeakerAccess")
@Subcomponent(modules = FragmentRootViewModule.class)
@FragmentScope
public interface AuroraChanceFragmentComponent {
	void inject(AuroraChanceFragment auroraChanceFragment);
}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/dagger/components/AuroraChanceFragmentComponent.kt
package se.gustavkarlsson.skylight.android.dagger.components

import dagger.Subcomponent
import se.gustavkarlsson.skylight.android.dagger.modules.ChancePresenterModule
import se.gustavkarlsson.skylight.android.dagger.modules.FragmentRootViewModule
import se.gustavkarlsson.skylight.android.dagger.modules.PresentingAuroraReportsModule
import se.gustavkarlsson.skylight.android.dagger.modules.TimeSinceUpdateControllerModule
import se.gustavkarlsson.skylight.android.dagger.scopes.FragmentScope
import se.gustavkarlsson.skylight.android.gui.activities.main.fragments.aurora_chance.AuroraChanceFragment

@Subcomponent(modules = arrayOf(
	FragmentRootViewModule::class,
	TimeSinceUpdateControllerModule::class,
	ChancePresenterModule::class,
	PresentingAuroraReportsModule::class
))
@FragmentScope
interface AuroraChanceFragmentComponent {
	fun inject(auroraChanceFragment: AuroraChanceFragment)
}
